File: bench/random/random.scala
/* The Great Computer Language Shootout 
   contributed by Isaac Gouy (Scala novice)

To compile:	
   scalac random.scala

To run:
   scala random 900000
*/


object random {
   def main(args: Array[String]) = {
      var n = toPositiveInt(args);
      var result: Double = _;

      while (n>0) { result=generate(100.0); n=n-1; }

      Console.println( result );
   }


   private val IM = 139968;
   private val IA = 3877;
   private val IC = 29573;
   private var seed = 42;

   def generate(max: Double) = {
      seed = (seed * IA + IC) % IM;
      max * seed / IM;
   }


   private def toPositiveInt(s: Array[String]) = {
      val i = 
         try { Integer.parseInt(s(0)); } 
         catch { case _ => 1 }
      if (i>0) i; else 1;
   }
}

File: bench/random/random.lua
--- The Great Computer Language Shootout
-- http://shootout.alioth.debian.org/
--
-- contributed by: Roberto Ierusalimschy
-- modified by Mike Pall

local IM = 139968
local IA = 3877
local IC = 29573

local LAST = 42
local mod = math.mod
local function gen_random(max)
    LAST = mod((LAST * IA + IC), IM)
    return( (max * LAST) / IM )
end

local N = tonumber((arg and arg[1])) or 1
local result = 0
for i=1, N do
    result = gen_random(100)
end
io.write(string.format("%.9f\n", result))
